### PR TITLE
Users/mitokic/hierarchical reconciliation bug

### PR DESCRIPTION
**Pull request by mitokic**
- fixed bug for standard and grouped hierarchical forecasts
- changed the best model output name from "Best Model" to "Best-Model" to follow similar formatting as other models
- automatically turned ensemble models off for yearly forecasts. In order to prevent multiple errors when building ensemble training data. Could be something to come back to and build a more robust solution going forward.

### CODE REVIEW

File: R/prepare_hierarchy_data.R [hunk 1]
```diff
@@ -4,13 +4,15 @@
...
+#' @param return return data or hierarchical ts object
 #' 
 #' @return data_tbl_final
 #' @noRd
 get_data_tbl_final <- function(data_tbl,
                           combo_variables,
                           forecast_approach,
-                          frequency_number){
+                          frequency_number, 
+                          return = "data"){
```
Comment (AKannanMSFT) [inline R/prepare_hierarchy_data.R]: Maybe we should call it return_type or something else

File: R/prepare_hierarchy_data.R [hunk 2]
```diff
@@ -128,23 +130,38 @@ get_data_tbl_final <- function(data_tbl,
...
+          hts::allts() %>%
+          data.frame() %>%
+          tibble::add_column(Date = Date,
+                             .before = 1)%>%
+          tidyr::pivot_longer(!Date, 
+                              names_to = "Combo", 
+                              values_to = "Target") %>%
+          tibble::tibble()
+        
+      } else if(return == "hts_gts") { # return a list of hierarchical info used when reconciling
```
Comment (AKannanMSFT) [inline R/prepare_hierarchy_data.R]: Ideally, we should have a function constructor as follows

``` r
data_cast_return <- function(df,ret_obj){
if(ret_obj == "data"){
          df %>%
         get_hts(some_list)  %>%
          hts::allts() %>%
          data.frame() %>%
          tibble::add_column(Date = Date,
                             .before = 1)%>%
          tidyr::pivot_longer(!Date, 
                              names_to = "Combo", 
                              values_to = "Target") %>%
          tibble::tibble()
} else if(return == "hts_gts") {
      df %>%
      get_hts(some_list)
} else{
     df
}
}

  data_cast %>%
          dplyr::select(-Date) %>%
          ts(frequency = frequency_number)%>% 
          data_cast_return(ret_obj)

```

**Review by AKannanMSFT — CHANGES_REQUESTED** [review]
There are some minor code structure edits. Please see if you'd like to address them.

**Comment by mitokic** [comment]
thanks for the feedback Aadharsh. I made the changes you requested and double checked they work for both grouped and standard hierarchies using test data. Please take another look and approve if everything looks good.

**Review by AKannanMSFT — APPROVED** [review]
Hierarchy Bug Fix

**Review by AKannanMSFT — DISMISSED** [review]
Bug fix by Mike

**Comment by AKannanMSFT** [comment]
Bug fix on heirarchy

**Pull request merged**